### PR TITLE
Sensor updates

### PR DESCRIPTION
**Pull request by 6by9**
Updates OV5647 fora load of required controls.
Minor fix to imx219.
Adds [G|S]_SELECTION support to Unicam.

### CODE REVIEW

**Comment by 6by9** [comment]
@naushir Need CID_VBLANK still.

**Comment by 6by9** [comment]
OK, I think this may be near complete. Please test and see what you get.

**Comment by 6by9** [comment]
Take it back. The exposure control range isn't being set to that defined by VBLANK - sorry but the orriginal patch adding exposure is just plain wrong.
Nearly there though.

File: drivers/media/i2c/ov5647.c [hunk 1]
```diff
@@ -613,65 +618,124 @@ static struct ov5647_mode supported_modes_10bit[] = {
...
 			.height = 1944
 		},
-		ov5647_2592x1944_10bit,
-		ARRAY_SIZE(ov5647_2592x1944_10bit)
+		.crop = {
+			.left = 0,
+			.top = 0,
+			.width = 2624,
+			.height = 1956,
+		},
```
Comment (naushir) [inline drivers/media/i2c/ov5647.c:639]: Should this not be 2592x1944?

Comment (6by9) [reply drivers/media/i2c/ov5647.c:639]: This was the mess I got into over native array vs active vs TIMING_x_OUTPUT_SIZE vs TIMING_x_ADDR_END-TIMING_x_ADDR_START vs what the heck was wanted by userspace.

Yes, 2592x1944 is probably correct.

**Comment by 6by9** [comment]
Updated. I think I'm done. Flips can wait for now.
@naushir If you can review and test, then say if you're happy.

**Comment by naushir** [comment]
LGTM

**Comment by davidplowman** [comment]
I've been testing this with my ov5647 and libcamera - looks good to me too.

**Comment by naushir** [comment]
@pelwell @popcornmix, this is ready to be merged if you are happy with it.

**Pull request merged**